db/src/internal_types.rs

```diff
@@ -181,6 +181,15 @@ impl TermWithTempIds {
     }
 }
 
+// TODO this seems like a sign that I'm doing something wrong...
```

Comment:
Not really.  I think this is just the `Into` implementation chaining `TermWithoutTempIds -> TermWithTempIds -> TermWithTempIdsAndLookupRefs`.  Break that out as a little Pre:, please.